pkg/types/powervs/validation/machinepool.go

```diff
@@ -1,27 +1,85 @@
-	/*
-		for i, zone := range p.Zones {
-			if !strings.HasPrefix(zone, platform.Region) {
-				allErrs = append(allErrs, field.Invalid(fldPath.Child("zones").Index(i), zone, fmt.Sprintf("Zone not in configured region (%s)", platform.Region)))
-			}
+
+	// Validate ServiceInstance
+	if !isUUID(p.ServiceInstance) {
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("ServiceInstance"), p.ServiceInstance, "Service Instance provided is not a UUID"))
+	}
+
+	// Validate Name
+	if !regexp.MustCompile(`^[a-zA-Z0-9-_]{1,}$`).MatchString(p.Name) {
+		allErrs = append(allErrs, field.Invalid(fldPath, p.Name, "Only letters (no accents), numbers, underscores and dashes are allowed"))
+	}
+	if len(p.Name) > 47 {
+		allErrs = append(allErrs, field.TooLong(fldPath, p.Name, 47))
+	}
+
+	// Validate VolumeIDs
+	for _, volumeID := range p.VolumeIDs {
+		if !isUUID(volumeID) {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("VolumeIDs"), p.VolumeIDs, "Volume ID provided is not a UUID"))
 		}
+	}
 
-		if p.OSDisk.DiskType != "" {
-			diskTypes := sets.NewString("pd-standard", "pd-ssd")
-			if !diskTypes.Has(p.OSDisk.DiskType) {
-				allErrs = append(allErrs, field.NotSupported(fldPath.Child("diskType"), p.OSDisk.DiskType, diskTypes.List()))
-			}
-		}*/
+	// Validate Memory
+	// TODO(mjturek): Check maximums on processors and memory. The maximum is a dynamic value
+	//                tied to sys type, service instance, and availability. We'd need to query
+	//                for the current maximum to handle this properly.
+	if p.Memory < 2 {
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("memory"), p.Memory, "Memory must be at least 2 GB"))
+	}
 
+	// Validqte Processors
+	if p.Processors < 0.25 {
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("processors"), p.Processors, "Number of processors must be at least .25"))
+	}
+	// TODO(mjturek): Issue a warning if using shared procType with non integer number of processors.
+	//                Processor count is set to next integer when using shared processors.
+	if math.Mod(float64(p.Processors), 0.25) != 0 {
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("processors"), p.Processors, "Processors must be in increments of .25"))
+	}
+
+	// Validate optional fields
+	// Validate ProcType
+	if p.ProcType != "" {
+		if p.ProcType != "shared" && p.ProcType != "dedicated" {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("procType"), p.ProcType, "ProcType must be either 'shared' or 'dedicated'."))
+		}
+	}
+
+	// Validate ImageID
+	if p.ImageID != "" {
+		if !isUUID(p.ImageID) {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("imageID"), p.ServiceInstance, "Image ID provided is not a UUID"))
+		}
+	}
+
+	for _, networkID := range p.NetworkIDs {
+		if !isUUID(networkID) {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("NetworkIDs"), p.VolumeIDs, "Network ID provided is not a UUID"))
+		}
+	}
+
+	sysTypes := sets.NewString("s922", "e980")
```

Comment:
In staging area system type is different, with this list installer doesn't work in the staging area.

Reply:
Moises promised that the name would always follow the pattern s922-*, so I think we can safely add this to the list (but a regex)

Reply:
I did a regex that accomplishes this but might be a pain if/when we update this list to include more systems. Still seems like the best solution.